modules/styling-transform/lib/getCssVariables.ts

```diff
@@ -0,0 +1,68 @@
+import {compile, Element} from 'stylis';
+
+export function getVariablesFromFiles(files: string[]): Record<string, string> {
+  return files.reduce((result, file) => {
+    extractVariables(compile(file)); //?
+    return {...result, ...extractVariables(compile(file))};
+  }, {} as Record<string, string>);
+}
+
+export function extractVariables(
+  blocks: Element[] | string,
+  variables: Record<string, string> = {}
+): Record<string, string> {
+  if (typeof blocks === 'string') {
+    return variables;
+  }
+  return blocks
+    .filter(block => block.type === 'rule')
+    .reduce(
+      (result, rule) => {
+        if (Array.isArray(rule.children)) {
+          rule.children.forEach(child => {
+            if (
+              typeof child.props === 'string' &&
+              child.props.startsWith('--') &&
+              typeof child.children === 'string'
+            ) {
+              result[child.props] = child.children;
+            }
+            if (Array.isArray(child.props)) {
+              //
```

Comment:
Should there be a comment here? Or should this be deleted?

Reply:
The types say it could be an array, but I didn't encounter it in our style dictionary CSS files